client/webserver/site/src/js/settings.ts

```diff
@@ -158,6 +157,15 @@ export default class SettingsPage extends BasePage {
     })
   }
 
+  /*
+   * slideSwap animates the replacement of the currently shown form with the
+   * newForm and sets this.currentForm.
+   */
+  slideSwap (newForm: PageElement) {
+    forms.slideSwap(this.currentForm, newForm)
```

Comment:
We should `await` here, no?

Reply:
I didn't think so. What's your reasoning?

Reply:
> I didn't think so. What's your reasoning?

`forms.slideSwap(..., ...)` returns a `Promise<void>`.

Reply:
After a close look at our use of `forms.slideSwap`, we definitely should `await`. `slideSwap` will have become an `async` function so we can `await` here. All current callers are `async` functions too but we don't need to `await` when we use `this.slideSwap` because we are not expecting a return value. Other use of `forms.slideSwap` should  `await` if they are used in `async` functions.

Reply:
You need `await` if you want to ensure that `this.currentForm = newForm` is only set after the animation is completed.